examples/webbilling-demo/src/tests/tax-calculation.test.ts

```diff
@@ -198,54 +199,89 @@ integrationTest.describe("Tax calculation", () => {
+          setTimeout(async () => {
+            route.continue();
+          }, 10_000);
+        } else {
+          route.continue();
+        }
       });
 
-      await enterEmail(page, email);
-      await enterCreditCardDetails(
-        page,
-        "4242 4242 4242 4242",
-        INVALID_CUSTOMER_DETAILS,
+      const italyTaxCalculationRequestPromise = page.waitForRequest(
+        (request) =>
+          request.url().includes("/calculate_taxes") &&
+          request.postDataJSON().country_code === "IT",
+      );
+      const newYorkTaxCalculationRequestPromise = page.waitForRequest(
+        (request) =>
+          request.url().includes("/calculate_taxes") &&
+          request.postDataJSON().country_code === "US",
       );
 
-      const skeleton = page.getByTestId("tax-loading-skeleton");
-      await expect(skeleton).toBeVisible();
+      const newYorkTaxCalculationResponsePromise = page.waitForResponse(
+        (response) =>
+          response.url().includes("/calculate_taxes") &&
+          response.request().postDataJSON().country_code === "US",
+      );
 
-      await expect(calculateTaxesCount).toBe(1);
+      const italyTaxCalculationResponsePromise = page.waitForResponse(
+        (response) =>
+          response.url().includes("/calculate_taxes") &&
+          response.request().postDataJSON().country_code === "IT",
+      );
 
-      // Clear the postal code, to make sure that the change event is triggered by Stripe
+      await enterEmail(page, email);
       await enterCreditCardDetails(
         page,
         "4242 4242 4242 4242",
         ITALY_CUSTOMER_DETAILS,
       );
 
+      await expect(taxSkeleton).toBeVisible();
+
+      // Must wait for the request to be started,
+      // visual skeleton is not enough to prevent race condition
+      await italyTaxCalculationRequestPromise;
+
+      await enterSecurityCode(page, "");
+
       await enterCreditCardDetails(
         page,
         "4242 4242 4242 4242",
         NEW_YORK_CUSTOMER_DETAILS,
       );
 
+      await expect(taxSkeleton).toBeVisible();
+
+      // Must wait for the request to be started,
+      // visual skeleton is not enough to prevent race condition
+      await newYorkTaxCalculationRequestPromise;
+
       await expect(page.getByText(/Sales Tax - New York/)).toBeVisible();
 
-      await expect(calculateTaxesCount).toBe(2);
+      await newYorkTaxCalculationResponsePromise;
 
-      await expect(
-        page.getByText(/We couldn't verify your billing address./i),
-      ).not.toBeVisible();
+      const hasCompleted = await Promise.race([
+        italyTaxCalculationResponsePromise.then(() => true),
+        new Promise((resolve) => setTimeout(() => resolve(false), 0)),
+      ]);
+
+      if (hasCompleted) {
+        expect("Italy tax calculation should not have finished").toBeFalsy();
+      } else {
+        expect("Italy tax calculation finished").toBeTruthy();
+      }
```

Comment:
This part seems hard to read for me, could we just fail if `firstRequestHasCompleted`?
```suggestion
      const firstRequestHasCompleted = await Promise.race([
        italyTaxCalculationResponsePromise.then(() => true),
        new Promise((resolve) => setTimeout(() => resolve(false), 0)),
      ]);
      
      if (firstRequestHasCompleted) {
        fail("Italy tax calculation should not have finished");
      }
```

Reply:
Looking into this with the new change altogether now.